src/qgsquick/qgsquickmapcanvasmap.cpp

```diff
@@ -258,14 +259,11 @@ void QgsQuickMapCanvasMap::setFreeze( bool freeze )
 
   mFreeze = freeze;
 
-  if ( !mFreeze && mNeedsRefresh )
+  if ( !mFreeze && mLastExtent != mapSettings()->visibleExtent() )
```

Comment:
@wonder-sk what do you think about this approach?
instead of enabling the change from the action (pan or zoom), it's automatically detected

Reply:
I thought the "needs refresh" flag is a bit more generic in case there are other properties that need to be taken into account but I'm easy...